docs/src/pages/index.mdx

```diff
@@ -0,0 +1,178 @@
+};
+
+decalre const USERS: User[];
+
+R.sortBy(
+  USERS,
+  [R.prop("lastLogin"), "desc"],
+  R.prop("lastName"),
+  R.prop("firstName"),
+);
+```
+
+## 👷 Fully Typed
+
+Libraries like Lodash and Ramda are written in JavaScript, with types added
+later. Remeda, on the other hand, is fully written in TypeScript! This means
+types are deeply integrated into the code, allowing us to shift the focus from
+runtime errors to compile-time errors, so you catch errors in your IDE and CI,
+not in your logs, **and not in production**.
+
+We strive to provide the best possible real-life types for all functions, often
+going beyond what is offered by the TypeScript library, so you don't need to use
+other libraries like 'type-fest' and 'ts-reset' just to get the types you
+expected.
+
+```ts
+const DATA = [1, 2, 3] as const;
+
+const builtin = DATA.map(DATA, (x) => x.toString());
+//    ^? string[]
+
+const withRemeda = R.map(DATA, (x) => x.toString());
+//    ^? [string, string, string]
+```
+
+## 🦾 Functional
+
+Functional programming is beneficial, making code more readable. However, there
+are situations where you don't need "pipes" and want to call just a single
+function. In Remeda, **all** functions have automatic currying built-in. This
+means the same imported function can be called either regularly (what we call
+_dataFirst_) or with the first parameter, _data_, omitted (what we call
+_dataLast_). When a dataLast invocation is detected, Remeda creates a curried
+version of that function on-the-fly, with all other parameters bound to it. This
+allows you to use the same function in both functional and imperative styles,
+without additional imports. It also means you can gradually adopt a functional
+style where it makes sense without compromising readability when it doesn't.
+
+```ts
+// Lodash
+_.pick(obj, ["firstName", "lastName"]);
+
+// Ramda
+R.pick(["firstName", "lastName"], obj);
+
+// In Remeda you get both!
+R.pick(obj, ["firstName", "lastName"]); // dataFirst
+R.pipe(obj, R.pick(["firstName", "lastName"])); // dataLast
+```
+
+In the above example, the "dataFirst" approach is more natural and programmer-
+friendly because when you type the second argument, you get auto-complete from
+the IDE. It's not possible to get auto-complete in Ramda because the data
+argument is not provided.
+
+The "dataLast" approach is helpful when writing data transformations, aka pipes.
+
+```ts
+const DATA = [
+  { name: "john", age: 20, gender: "m" },
+  { name: "marry", age: 22, gender: "f" },
+  { name: "samara", age: 24, gender: "f" },
+  { name: "paula", age: 24, gender: "f" },
+  { name: "bill", age: 33, gender: "m" },
+];
+
+// Remeda
+R.pipe(
+  DATA,
+  R.filter((x) => x.gender === "f"),
+  R.groupBy((x) => x.age),
+);
+
+// Ramda
+R.pipe(
+  R.filter((x) => x.gender === "f"),
+  R.groupBy((x) => x.age),
+)(DATA); // broken typings in TS :(
+
+// Lodash
+_(DATA)
+  .filter((x) => x.gender === "f")
+  .groupBy((x) => x.age)
+  .value();
```

Comment:
**question:** hm, why is this example here? the types are all correct in this case, even the result type